doc/sphinx/addendum/implicit-coercions.rst

```diff
@@ -56,51 +56,40 @@ A name ``f`` can be declared as a coercion between a source user-defined class
 conditions holds:
 
  * ``D`` is a user-defined class, then the type of ``f`` must have the form
-   :g:`forall (x₁:A₁)..(xₙ:Aₙ)(y:C x₁..xₙ), D u₁..uₘ` where :math:`m`
+   :g:`forall (x₁:A₁)..(xₖ:Aₖ)(y:C v₁..vₙ), D u₁..uₘ` where :math:`m`
    is the number of parameters of ``D``.
  * ``D`` is ``Funclass``, then the type of ``f`` must have the form
-   :g:`forall (x₁:A₁)..(xₙ:Aₙ)(y:C x₁..xₙ)(x:A), B`.
+   :g:`forall (x₁:A₁)..(xₖ:Aₖ)(y:C v₁..vₙ)(x:A), B`.
  * ``D`` is ``Sortclass``, then the type of ``f`` must have the form
-   :g:`forall (x₁:A₁)..(xₙ:Aₙ)(y:C x₁..xₙ), s` with ``s`` a sort.
+   :g:`forall (x₁:A₁)..(xₖ:Aₖ)(y:C v₁..vₙ), s` with ``s`` a sort.
 
-We then write :g:`f : C >-> D`. The restriction on the type
-of coercions is called *the uniform inheritance condition*.
+We then write :g:`f : C >-> D`.
+
+.. _ambiguous-paths:
+
+When a coercion is added to the inheritance graph, new coercion paths
+which have the same classes as existing ones are ignored. The
+convertibility of the new and existing paths is checked and a warning
+is displayed when the check fails. When the :g:`x₁..xₖ` are exactly
```

Comment:
```suggestion
When you declare a new coercion (e.g. with :cmd:`Coercion`), new coercion
paths with the same classes as existing ones are ignored.  Coq will generate
a warning.  When the :g:`x₁..xₖ` are exactly
```

I've assumed that "The convertibility of the new and existing paths is checked and a warning is displayed when the check fails." merely restates the first sentence.  Is that correct?  If not, can you be specific about "is checked ... check fails"--what is the check?

Reply:
Not exactly, the check is a convertibility check (paths that are convertible (i.e., the same modulo Coq calculus) are acceptable, others will very likely cause troubles). Is it clearer with the new wording?

Reply:
That's better, let's go with this for now.  We can do better when we more thoroughly revise the chapter more thoroughly.